File: CS50git/cs50/pset6/less/caesar.py
# Usage: key
# Prompt user for text to cipher
# Cipher through caesar algorithm with key
# Return ciphered text
import sys
import string


def main(argv):

    # Init
    ALPHA = 26
    UPPER = 65
    LOWER = 97
    MOD = 32
    key = 0

    # Verify usage
    if len(sys.argv) != 2:
        print("Usage: key")
        exit(1)
    try:
        key = int(argv[1])
    except ValueError:
        print("Usage: key")
        exit(1)

    # Prompt user text to cipher
    plain_text = input("plaintext: ")

    # Output ciphered text
    print("ciphertext: ", end="")
    for char in plain_text:
        # char -> index -> ciphered index -> ciphered char
        if char.isalpha():
            ciphered_index = (ord(char) % MOD) - 1
            ciphered_index = (ciphered_index + key) % ALPHA
            if char.islower():
                char = chr(ciphered_index + LOWER)
            else:
                char = chr(ciphered_index + UPPER)
        print(char, end="")
    print()

    exit(0)


if __name__ == "__main__":
    main(sys.argv)

File: CS50git/cs50/pset2/caesar/caesar.c
// 18/11/2019
// Caesar

// Take for command-line argument an integer
// Prompt user for a text
// Cipher and return the text

#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>

string ciphering(int key, string plainText);

int main(int argc, char *argv[])
{
    // Verify usage and store key
    if (argc != 2)
    {
        fprintf(stdout, "Usage: integer\n");
        return 1;
    }
    int key = 0;
    if (!sscanf(argv[1], "%i", &key))
    {
        fprintf(stdout, "Usage: integer\n");
        return 2;
    }

    // Prompt user for plain text
    string plainText = get_string("Text: ");

    // Print ciphered text
    fprintf(stdout, "ciphertext: %s\n", ciphering(key, plainText));

    return 0;
}

// Cipher function
string ciphering(int key, string plainText)
{
    // Init required variables
    int alphabeticalIndex = 0;
    int asciiMinUpper = 65;
    int asciiMinLower = 97;

    // Cipher formula is ci = (pi + k) % 26

    for (int i = 0, n = strlen(plainText); i < n; i++)
    {
        if (isalpha(plainText[i]))
        {
            // Cipher uppercase letter
            if (isupper(plainText[i]))
            {
                alphabeticalIndex = (int) plainText[i] - asciiMinUpper;
                alphabeticalIndex = (alphabeticalIndex + key) % 26;
                plainText[i] = alphabeticalIndex + asciiMinUpper;

            }
            // Cipher lowercase letter
            else
            {
                alphabeticalIndex = (int) plainText[i] - asciiMinLower;
                alphabeticalIndex = (alphabeticalIndex + key) % 26;
                plainText[i] = alphabeticalIndex + asciiMinLower;
            }
        }
    }
    return plainText;
}